Vuelidate

BJordi opened the pull request:
## Summary

- Added Vuelidate library.
- Added validations to inputs in login form.
- Added functions to validate numbers and capital letters.
- Minor fixes in classes.

## Screenshot

![form-validations](https://user-images.githubusercontent.com/22583914/51004557-f9279f80-1519-11e9-9723-7574315971b5.gif)

Commit:
Added Vuelidate

Commit:
Added functions to validate numbers and capital letters.

Commit:
Fixed and added classes for wrong and valid inputs and labels.

Commit:
Added validations to login form.

FrankIglesias commented on src/main.js:
```diff
@@ -1,9 +1,11 @@
 import Vue from 'vue'
+import Vuelidate from 'vuelidate'
```
leave an empty line after this

FrankIglesias commented on src/scss/commons/colors.scss:
```diff
@@ -2,5 +2,6 @@ $green: #A6CC45;
 $light-grey: #F4F4F4;
 $light-grey2: #F6F6F6;
 $light-grey3: #EFF0E9;
+$red: #CC0000;
```
#C00

FrankIglesias commented on src/scss/commons/components.scss:
```diff
@@ -35,4 +37,18 @@
+  border: 1px solid $red;
+}
+
+.valid-input {
```
I think this is not part of the design and adds a lot of overhead to the class logic

BJordi replied:
Ok, I'll remove it! :ok_hand:

FrankIglesias commented on src/views/Login.vue:
```diff
@@ -1,40 +1,99 @@
+        <span class="error-label" v-if="!$v.email.email">El email no tiene un formato válido</span>
+        <span class="error-label" v-if="!$v.email.required && submitted">Campo requerido</span>
+      </div>
+      <!-- Password -->
```
remove comment

BJordi replied:
As it's a lot of code all together, I thought it was a good idea to separate each "section" with a comment.

FrankIglesias commented on src/views/Login.vue:
```diff
@@ -1,40 +1,99 @@
+          v-if="$v.password.$dirty && !$v.password.hasUppercase"
+        >El password debe contener al menos una mayúscula</span>
+      </div>
+
```
remove line

FrankIglesias commented on src/views/Login.vue:
```diff
@@ -1,40 +1,99 @@
 </template>
 
 <script>
+import { required, minLength, email } from "vuelidate/lib/validators";
```
remove semi colon

Commit:
Minor fixes: Added empty line. Removed comments, semicolons, and empty lines. Fixed color.

Commit:
Added wolox-equalizer.

Commit:
Fixes in classes and added wolox-equalizer.

Commit:
Removed logic for valid inputs.

FrankIglesias commented on src/scss/application.scss:
```diff
@@ -1 +1,10 @@
+  min-height: 100vh;
+  justify-content: center;
+  text-align: center;
+}
```
add EOL

FrankIglesias commented on src/views/Login.vue:
```diff
@@ -1,57 +1,101 @@
+        <input
+          id="email"
+          v-model="email"
+          :class="['input', ($v.email.$invalid && submitted) || !$v.email.email ? 'error-input' : '']"
```
`'error-input' : ($v.email.$invalid && submitted) || !$v.email.email)`

FrankIglesias replied:
why not only `'error-input' : ($v.email.$invalid && submitted)`?

BJordi replied:
Because while it's being written and it doesn't have the format of an email, I want it to appear in red and `!$v.email.email` checks that =)

BJordi replied:
Ok, but if I do what you say, I will have to add `'input': true`

FrankIglesias commented on src/views/Login.vue:
```diff
@@ -1,57 +1,101 @@
+          id="password"
+          type="password"
+          v-model="$v.password.$model"
+          :class="['input', $v.password.$error && ($v.password.required || submitted ) ? 'error-input': '']"
```
same

FrankIglesias commented on src/views/Login.vue:
```diff
@@ -1,57 +1,101 @@
+    },
+    email: {
+      required,
+      email
```
use another name for this like validEmailFormat

BJordi replied:
The one in the line 73 is the specific argument I have to use according to Vuelidate documentation.

BJordi replied:
![screenshot from 2019-01-14 22-51-48](https://user-images.githubusercontent.com/22583914/51153456-06f95f80-184f-11e9-8c38-c98b09c202ee.png)

FrankIglesias commented on src/views/Login.vue:
```diff
@@ -1,57 +1,101 @@
-      console.log(this.generateObject());
+    onSubmit () {
+      this.submitted = true
+      console.log(this.generateObject())
```
remove this console log

BJordi replied:
But it is part of what the card asks:
![screenshot from 2019-01-14 22-19-42](https://user-images.githubusercontent.com/22583914/51151730-994b3480-184a-11e9-8ab2-f030616a3498.png)

Commit:
Replace ternary operator by boolean objects in class binding.

Commit:
Added EOL and empty line after import.

Commit:
Solved conflicts with 'login-layout'

FrankIglesias commented on src/scss/commons/components.scss:
```diff
@@ -24,10 +25,15 @@
   display: flex;
   justify-content: center;
+
+  &:focus, &:active {
```
&:focus,
&:active

FrankIglesias commented on src/views/Login.vue:
```diff
@@ -1,57 +1,101 @@
-      <input id="name" class="input" v-model="password" type="password">
-    </div>
-    <button class="primaryButton" @click="printForm">Sign up</button>
+    <form @submit.prevent="onSubmit">
```
please migrate this to a pug

BJordi replied:
As I mentioned in another comments, I will do it in a specific PR.

FrankIglesias commented on src/views/Login.vue:
```diff
@@ -1,57 +1,101 @@
+    </form>
     <div class="horizontal-division"/>
-    <a :class="'secondaryButton'">Login</a>
+    <a class="secondary-button">Login</a>
```
use a router-link

BJordi replied:
This has been corrected in another PR but will do it again in it.

mcallegari10 commented on src/views/Login.vue:
```diff
@@ -1,57 +1,101 @@
 <template>
```
Use pug

BJordi replied:
Yeah, as I mentioned in another comment, I will migrate all files in one specific PR.

Commit:
Minor fixes
- Separated pseudo-classes into different lines
- Replaced anchor tag by router-link
- Added .env to gitignore
